classes/UpgradeSelfCheck.php

```diff
@@ -181,24 +132,133 @@ class UpgradeSelfCheck
+    const KEY_GENERATION_INVALID = 14;
+    const NOT_WRITING_DIRECTORY_LIST_NOT_EMPTY = 15;
+    const SHOP_VERSION_NOT_MATCHING_VERSION_IN_DATABASE = 16;
+
+    // Warnings const
+    const MODULE_VERSION_IS_OUT_OF_DATE = 17;
+    const PHP_REQUIREMENTS_UNKNOWN = 18;
+    const TEMPERED_FILES_LIST_NOT_EMPTY = 19;
+
+    const PHP_REQUIREMENTS_VALID = 20;
 
     public function __construct(
         Upgrader $upgrader,
         PrestashopConfiguration $prestashopConfiguration,
+        Translator $translator,
         DistributionApiService $distributionApiService,
+        ChecksumCompare $checksumCompare,
         string $prodRootPath,
         string $adminPath,
-        string $autoUpgradePath
+        string $autoUpgradePath,
+        string $originVersion
     ) {
         $this->upgrader = $upgrader;
         $this->prestashopConfiguration = $prestashopConfiguration;
+        $this->translator = $translator;
         $this->distributionApiService = $distributionApiService;
+        $this->checksumCompare = $checksumCompare;
         $this->prodRootPath = $prodRootPath;
         $this->adminPath = $adminPath;
         $this->autoUpgradePath = $autoUpgradePath;
+        $this->originVersion = $originVersion;
+    }
+
+    /**
+     * @throws Exception
+     *
+     * @return array<int, bool>
+     */
+    public function getErrors(): array
+    {
+        return [
+            self::PHP_REQUIREMENTS_INVALID => $this->getPhpRequirementsState() == self::PHP_REQUIREMENTS_INVALID,
+            self::ROOT_DIRECTORY_NOT_WRITABLE => !$this->isRootDirectoryWritable(),
+            self::ADMIN_UPGRADE_DIRECTORY_NOT_WRITABLE => !$this->isAdminAutoUpgradeDirectoryWritable(),
+            self::SAFE_MODE_ENABLED => !$this->isSafeModeDisabled(),
+            self::F_OPEN_AND_CURL_DISABLED => !$this->isFOpenOrCurlEnabled(),
+            self::ZIP_DISABLED => !$this->isZipEnabled(),
+            self::MAINTENANCE_MODE_DISABLED => !$this->isLocalEnvironment() || !$this->isShopDeactivated(),
+            self::CACHE_ENABLED => !$this->isCacheDisabled(),
+            self::MAX_EXECUTION_TIME_VALUE_INCORRECT => $this->getMaxExecutionTime() > 0 && $this->getMaxExecutionTime() < 30,
+            self::APACHE_MOD_REWRITE_DISABLED => !$this->isApacheModRewriteEnabled(),
+            self::NOT_LOADED_PHP_EXTENSIONS_LIST_NOT_EMPTY => !empty($this->getNotLoadedPhpExtensions()),
+            self::NOT_EXIST_PHP_FUNCTIONS_LIST_NOT_EMPTY => !empty($this->getNotExistsPhpFunctions()),
+            self::MEMORY_LIMIT_INVALID => !$this->isMemoryLimitValid(),
+            self::PHP_FILE_UPLOADS_CONFIGURATION_DISABLED => !$this->isPhpFileUploadsConfigurationEnabled(),
+            self::KEY_GENERATION_INVALID => !$this->checkKeyGeneration(),
+            self::NOT_WRITING_DIRECTORY_LIST_NOT_EMPTY => !empty($this->getNotWritingDirectories()),
+            self::SHOP_VERSION_NOT_MATCHING_VERSION_IN_DATABASE => !$this->isShopVersionMatchingVersionInDatabase(),
+        ];
```

Comment:
If we're supposed to return only display errors, this method could return only the detected issues. Something OK would not appear.

And if you only return the issues, you could generate at this time the translated message? By doing so you would call trans only when necessary.